example/ram/compile.lurk

```diff
@@ -0,0 +1,71 @@
+(defmacro or (a b)
+  `(if ,a t ,b))
+
+(define map
+  (lambda (f xs)
+    (if (eq nil xs)
+        nil
+        (cons (f (car xs)) (map f (cdr xs))))))
+
+(define cadr
+  (lambda (xs)
+    (car (cdr xs))))
+
+(define caddr
+  (lambda (xs)
+    (car (cdr (cdr xs)))))
+
+(define cadddr
+  (lambda (xs)
+    (car (cdr (cdr (cdr xs))))))
+
+(define let-bindings cadr)
+(define let-body caddr)
+(define lambda-params cadr)
+(define lambda-body caddr)
+
+(define lookup
+  (lambda (e env success failure)
+    (if (eq nil env)
+        (failure)
+        (if (eq nil (atom (car (car env))))
+            (lookup e (car env) success
+                    (lambda () (lookup e (cdr env) success failure)))
+            (if (eq e (car (car env)))
+                (success (cdr (car env)))
+                (lookup e (cdr env) success failure))))))
+
+(define member
+  (lambda (e xs)
+    (if (eq nil xs)
+        nil
+        (if (eq e (car xs))
+            xs
+            (member e (cdr xs))))))
+
+(define compile-closure
+  (lambda (e params ram env)
+    (let ((init-params (append `(if current-env current-ram eval begin atom car cdr emit quote macroexpand + - / * = eq cons) params))
+          (rec (lambda (e) (compile-closure e init-params ram env)))
+          (rec-params (lambda (e params) (compile-closure e (append init-params params) ram env))))
+      (if (atom e)
+          (if (member e params)
+              e
+              (lookup e env (lambda (v)
+                              ;;(if (functionp v) e v)
```

Comment:
What is the significance of this comment? Should it be removed, or re-enabled, or…?

Reply:
If removed, is the `success` arg to `lookup` needed at all? It seems like we always pass the identity function.

Reply:
I was experimenting with not inlining function values at first. I don't think we'd want to make the `success` arg implicit, as it would obscure the flow.